haystack/preview/document_stores/mixins.py

```diff
@@ -0,0 +1,67 @@
+from typing import Protocol, Dict, Optional
+
+from haystack.preview.document_stores.protocols import Store
+
+
+class StoreComponent(Protocol):
+    """
+    Represents a component that needs a single store to run.
+    """
+
+    _store: Store
+
+    @property
+    def store(self) -> Optional[Store]:
+        ...
+
+    @store.setter
+    def store(self, store: Store):
+        ...
+
+
+class StoreMixin(StoreComponent):
+    """
+    Adds the capability of a component to use a single document store from the `self.store` property.
+    """
+
+    @property
+    def store(self) -> Optional[Store]:
+        return getattr(self, "_store", None)
```

Comment:
```suggestion
    self._store: Store = None

    @property
    def store(self) -> Optional[Store]:
        return self._store

Reply:
Ah yeah, bad copy and paste. I meant to have this:

```suggestion
    _store: Store = None

    @property
    def store(self) -> Optional[Store]:
        return self._store

Reply:
Aren't you defining a class attribute that way?

Reply:
If you access it from the class it is, but all instances of classes that inherit this mixin will store their own value.

```
In [1]: class Foo:
   ...:     value: int = 10
   ...:

In [2]: Foo.value
Out[2]: 10

In [3]: f = Foo()

In [4]: b = Foo()

In [5]: f.value, b.value
Out[5]: (10, 10)

In [6]: f.value = 100

In [7]: f.value, b.value
Out[7]: (100, 10)

In [8]: Foo.value
Out[8]: 10
```